backends/smt2/smt2.cc

```diff
@@ -824,38 +824,49 @@ struct Smt2Worker
 					is_register = true;
 			if (wire->port_id || is_register || wire->get_bool_attribute(ID::keep) || (wiresmode && wire->name[0] == '\\')) {
 				RTLIL::SigSpec sig = sigmap(wire);
+				std::string comment = "";
 				if (wire->port_input)
-					decls.push_back(stringf("; yosys-smt2-input %s %d\n", get_id(wire), wire->width));
+					comment = stringf("; yosys-smt2-input %s %d\n", get_id(wire), wire->width);
 				if (wire->port_output)
-					decls.push_back(stringf("; yosys-smt2-output %s %d\n", get_id(wire), wire->width));
+					comment = stringf("; yosys-smt2-output %s %d\n", get_id(wire), wire->width);
 				if (is_register)
-					decls.push_back(stringf("; yosys-smt2-register %s %d\n", get_id(wire), wire->width));
+					comment = stringf("; yosys-smt2-register %s %d\n", get_id(wire), wire->width);
 				if (wire->get_bool_attribute(ID::keep) || (wiresmode && wire->name[0] == '\\'))
-					decls.push_back(stringf("; yosys-smt2-wire %s %d\n", get_id(wire), wire->width));
+					comment = stringf("; yosys-smt2-wire %s %d\n", get_id(wire), wire->width);
 				if (GetSize(wire) == 1 && (clock_posedge.count(sig) || clock_negedge.count(sig)))
-					decls.push_back(stringf("; yosys-smt2-clock %s%s%s\n", get_id(wire),
-							clock_posedge.count(sig) ? " posedge" : "", clock_negedge.count(sig) ? " negedge" : ""));
+					comment = stringf("; yosys-smt2-clock %s%s%s\n", get_id(wire),
+							clock_posedge.count(sig) ? " posedge" : "", clock_negedge.count(sig) ? " negedge" : "");
 				if (bvmode && GetSize(sig) > 1) {
+					std::string sig_bv = get_bv(sig);
+					if (!comment.empty())
+						decls.push_back(comment);
 					decls.push_back(stringf("(define-fun |%s_n %s| ((state |%s_s|)) (_ BitVec %d) %s)\n",
-							get_id(module), get_id(wire), get_id(module), GetSize(sig), get_bv(sig).c_str()));
+							get_id(module), get_id(wire), get_id(module), GetSize(sig), sig_bv.c_str()));
 					if (wire->port_input)
 						ex_input_eq.push_back(stringf("  (= (|%s_n %s| state) (|%s_n %s| other_state))",
 								get_id(module), get_id(wire), get_id(module), get_id(wire)));
 				} else {
-					for (int i = 0; i < GetSize(sig); i++)
+					std::string sig_bool[GetSize(sig)];
```

Comment:
This apparently isn't standard C++ and isn't supported in older versions of clang (cf the CI failure), but you should be able to use a `std::vector<std::string>` to the same effect.